std/traits.d

```diff
@@ -3732,19 +3740,27 @@ template hasElaborateAssign(S)
     static assert( hasElaborateAssign!SS9);
 }
 
-/**
-   True if `S` or any type directly embedded in the representation
-   of `S` defines an elaborate destructor. Elaborate destructors
-   are introduced by defining `~this()` for a $(D
-   struct).
+version (StdDdoc)
```

Comment:
Is there any reason why we should have essentially the same code reaped twice? I suggest keeping only the longer syntax versions (with explicit `template`) as they can be documented (while the aliases can't IIRC).

Reply:
I believe the answer to that question can be found in the previous discussion in this PR. The alias symbol version is added to halve the number of template instances and the eponymous template is needed for the sake of ddoc. Using 

```d
template Unqual(S) {}
```

and similar for the other four templates, as @andralex proposed previously, crashes the `CyberShadow/DAutoTest` CI job for some unknown reason.

Reply:
I believe the answer to that question can be found in the title and previous discussions of this PR. The alias symbol version is added to halve the number of template instances being the main directive of this change. The eponymous template version is needed for the sake of ddoc. Opposite to using 

```d
template Unqual(S) {}
```

and similarly for the other four templates, as @andralex proposed previously, that crashes the `CyberShadow/DAutoTest` CI job for some unknown reason.